tck/src/test/java/org/eclipse/microprofile/jwt/tck/util/TokenUtilsGenerateTokenTest.java

```diff
@@ -0,0 +1,152 @@
+        invalidFields.add(TokenUtils.InvalidClaims.SIGNER);
+        String token = TokenUtils.generateTokenString("/Token1.json", invalidFields);
+        validateToken(token);
+    }
+
+    @Test(groups = TCKConstants.TEST_GROUP_UTILS, expectedExceptions = {InvalidJwtException.class},
+        description = "Illustrate validation of exp")
+    public void testFailExpired() throws Exception {
+        HashMap<String, Long> timeClaims = new HashMap<>();
+        HashSet<TokenUtils.InvalidClaims> invalidFields = new HashSet<>();
+        invalidFields.add(TokenUtils.InvalidClaims.EXP);
+        String token = TokenUtils.generateTokenString("/Token1.json", invalidFields, timeClaims);
+        validateToken(token);
+    }
+
+    @Test(groups = TCKConstants.TEST_GROUP_UTILS, expectedExceptions = {InvalidJwtException.class},
+        description = "Illustrate validation of exp that has just expired")
+    public void testFailJustExpired() throws Exception {
+        HashMap<String, Long> timeClaims = new HashMap<>();
+        // Set exp to 61 seconds in past
+        long exp = TokenUtils.currentTimeInSecs() - 61;
+        timeClaims.put(Claims.exp.name(), exp);
+        String token = TokenUtils.generateTokenString("/Token1.json", null, timeClaims);
+        validateToken(token);
+    }
+
+    @Test(groups = TCKConstants.TEST_GROUP_UTILS,
+        description = "Illustrate validation of exp that is in grace period")
+    public void testExpGrace() throws Exception {
+        HashMap<String, Long> timeClaims = new HashMap<>();
+        // Set exp to 45 seconds in past
+        long exp = TokenUtils.currentTimeInSecs() - 45;
+        timeClaims.put(Claims.exp.name(), exp);
+        String token = TokenUtils.generateTokenString("/Token1.json", null, timeClaims);
+        validateToken(token);
+    }
+
+    private void validateToken(String token) throws Exception {
+
+        RSAPublicKey publicKey = (RSAPublicKey) TokenUtils.readPublicKey("/publicKey.pem");
+        int expGracePeriodSecs = 60;
+
+        JwtConsumerBuilder builder = new JwtConsumerBuilder();
+
+        // 'exp' must be available
+        builder.setRequireExpirationTime();
+        // 'iat' must be available
+        builder.setRequireIssuedAt();
+        // 'RS256' is required
+        builder.setJwsAlgorithmConstraints(
+           new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST, "RS256"));
+
+        // issuer must be equal to TCKConstants.TEST_ISSUER
+        builder.setExpectedIssuer(true, TCKConstants.TEST_ISSUER);
+        builder.setVerificationKey(publicKey);
+        builder.setAllowedClockSkewInSeconds(expGracePeriodSecs);
+
+        JwtClaims claimsSet = builder.build().processToClaims(token);
+        // Confirm all the claims available in /Token1.json have made it into the verified claimSet
+
+        Assert.assertEquals(claimsSet.getClaimsMap().size(), 18);
+        Assert.assertEquals("https://server.example.com", claimsSet.getIssuer());
```

Comment:
The signature is `Assert.assertEquals(Actual, Expected)` and thus parameters are swapped (also for the following lines)